ipmininet/router/config/openr.py

```diff
@@ -47,14 +50,29 @@ def __init__(self, node, *args, **kwargs):
 
     def build(self):
         cfg = super().build()
+        self.options.redistribute_ifaces = \
+            ','.join([intf.name for intf in self._node.intfList()])
         cfg.update(self.options)
-        cfg.node_name = self._node.name
+        cfg.redistribute_ifaces = \
+        self._mklogdirs()
         interfaces = realIntfList(self._node)
         cfg.interfaces = self._build_interfaces(interfaces)
         cfg.networks = self._build_networks(interfaces)
         cfg.prefixes = self._build_prefixes(interfaces)
         return cfg
 
+    def _mklogdirs(self):
```

Comment:
Given that you create a temp dir for the whole node (i.e., /tmp/<node name>) would it make sense to put this creation code in the node itself? e.g., before the config build process in IPNode: https://github.com/cnp3/ipmininet/blob/master/ipmininet/router/__router.py#L108

This would encourage later refactoring of the daemons to put everything in this node-specific directory.

Reply:
I think, also with respect to your request to expand the logdir creation to more daemons, it should be a feature of the generic daemon class, since `logfile` is a daemon attribute: https://github.com/cnp3/ipmininet/blob/master/ipmininet/router/config/base.py#L298.

It could be also happen in the `Router` class, because here we have access to the config which holds the `logfile` option. https://github.com/cnp3/ipmininet/blob/master/ipmininet/router/__router.py#L189. There I can overwrite the IPNode start method and it could happen when the node starts up.